pandas/core/categorical.py

```diff
@@ -294,22 +293,10 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
 
         # sanitize input
         if is_categorical_dtype(values):
+            if dtype.categories is None:
+                dtype = CategoricalDtype(values.categories, dtype.ordered)
 
-            # we are either a Series or a CategoricalIndex
-            if isinstance(values, (ABCSeries, ABCCategoricalIndex)):
-                values = values._values
-
-            if ordered is None:
-                ordered = values.ordered
-            if categories is None:
-                categories = values.categories
-            values = values.get_values()
-
-        elif isinstance(values, (ABCIndexClass, ABCSeries)):
-            # we'll do inference later
-            pass
-
-        else:
+        elif not isinstance(values, (ABCIndexClass, ABCSeries)):
 
             # on numpy < 1.6 datetimelike get inferred to all i8 by
```

Comment:
you can remove this commnet (about numpy 1.6); also I think we can eliminate this logic (here or antother PR) as we don't care about old numpy any longer